### PR TITLE
ref(integrationfeature): Drop sentry_app_id column

### PR DESCRIPTION
**Pull request by ceorourke**
This is step 6 of the migration plan [here](https://github.com/getsentry/sentry/pull/30521) to update the `IntegrationFeature` table - now we actually drop the `sentry_app_id` column.

### CODE REVIEW

File: src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        ("sentry", "0261_prepare_remove_sentry_app_column"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            state_operations=[],
+            database_operations=[
+                migrations.RunSQL(
+                    sql="ALTER TABLE sentry_integrationfeature DROP COLUMN sentry_app_id",
+                    reverse_sql="ALTER TABLE sentry_integrationfeature ADD COLUMN sentry_app_id int NULL",
```
Comment (AniketDas-Tekky) [inline src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py:37]: If the migration rolls back then we won't be able to populate old `sentry_app_id`'s right?

Comment (ceorourke) [reply src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py:37]: Yeah I wasn't actually sure what to put here, cause they'll be lost so I just went with nothing 🤷

Comment (AniketDas-Tekky) [reply src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py:37]: In the past, I renamed a column to something like `sentry_app_id_deprecated` and on the roll back would just rename it back.

Comment (wedamija) [reply src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py:37]: The rollback is just a courtesy for dev environments, it is never used in production. We typically do the bare minimum here to allow migrations to work forwards/backwards.

**Review by wedamija — APPROVED** [review]

File: src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply src/sentry/migrations/0262_drop_sentry_app_from_integrationfeature.py:37]: then LGTM :)

**Pull request merged**